test/DurableTask.SqlServer.AzureFunctions.Tests/TargetBasedScalingTests.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace DurableTask.SqlServer.AzureFunctions.Tests
+{
+    using System;
+    using DurableTask.Core;
+    using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+    using Microsoft.Azure.WebJobs.Host.Scale;
+    using Microsoft.Extensions.Logging;
+    using Moq;
+    using Xunit;
+    using Xunit.Abstractions;
+
+    public class TargetBasedScalingTests
+    {
+        readonly Mock<SqlMetricsProvider> metricsProviderMock;
+        readonly Mock<IOrchestrationService> orchestrationServiceMock;
+
+        public TargetBasedScalingTests()
+        {
+            this.orchestrationServiceMock = new Mock<IOrchestrationService>(MockBehavior.Strict);
+
+            this.metricsProviderMock = new Mock<SqlMetricsProvider>(
+                MockBehavior.Strict,
+                null,
+                null);
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(10)]
+        [InlineData(20)]
+        public async void TargetBasedScalingTest(int expectedTargetWorkerCount)
```

Comment:
is it theoretically possible for the expectedWorkerCount to return a negative number? I guess that the `int` type doesn't have any prevention against it. In the spirit of defensive programming, should we have a guard against that edge case?

Reply:
Shouldn't be possible, based on the code [here](https://github.com/microsoft/durabletask-mssql/blob/6003e41398779f7796aead207e399d3031c6e623/src/DurableTask.SqlServer/Scripts/logic.sql#L147-L167). But I'm not opposed to adding some defensive code.